src/test/ui/transmutability/abstraction/const_generic_fn.rs

```diff
@@ -0,0 +1,41 @@
+// check-pass
+//! An array must have the correct length.
+
+#![crate_type = "lib"]
+#![feature(transmutability)]
+#![allow(dead_code, incomplete_features, non_camel_case_types)]
+
+mod assert {
+    use std::mem::BikeshedIntrinsicFrom;
+    pub struct Context;
+
+    pub fn array_like<T, E, const N: usize>()
+    where
+        T: BikeshedIntrinsicFrom<[E; N], Context, false, false, false, true>,
+        [E; N]: BikeshedIntrinsicFrom<T, Context, false, false, false, true>
```

Comment:
Add tests where some of these arguments are associated consts of generic params or const generics

Reply:
Resolved by https://github.com/rust-lang/rust/pull/92268/commits/ed0e23c8d8d271d507a0576ac69d1a4494751e6c